### PR TITLE
commit-mono: 1.136 -> 1.138

### PR DESCRIPTION
**Pull request by zzzsyyy**
## Description of changes

Diff: https://github.com/eigilnikolajsen/commit-mono/compare/1.136...1.138
Changelogs: 
https://github.com/eigilnikolajsen/commit-mono/releases/tag/1.137
https://github.com/eigilnikolajsen/commit-mono/releases/tag/1.138

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by moonshxne — COMMENTED** [review]
Would it be better to be more explicit, e.g., `install -Dm644 -t $out/share/fonts/truetype CommitMono-${version}/ttfautohint/*.ttf`?

**Comment by moonshxne** [comment]
@yoav-lavi A reminder to please review this commit? Looking forward to using the latest Commit Mono from `nixpkgs`!

File: pkgs/data/fonts/commit-mono/default.nix [hunk 1]
```diff
@@ -20,7 +20,8 @@ stdenvNoCC.mkDerivation rec {
 
   installPhase = ''
     runHook preInstall
-    install -Dm644 -t $out/share/fonts/opentype/ *.otf
+    install -Dm644 -t $out/share/fonts/opentype/ CommitMono-${version}/*.otf
```
Comment (yoav-lavi) [inline pkgs/data/fonts/commit-mono/default.nix:23]: Why is the version needed here?

File: pkgs/data/fonts/commit-mono/default.nix [hunk 2]
```diff
@@ -20,7 +20,8 @@ stdenvNoCC.mkDerivation rec {
...
+    install -Dm644 -t $out/share/fonts/truetype/ CommitMono-${version}/ttfautohint/*.ttf
```
Comment (yoav-lavi) [inline pkgs/data/fonts/commit-mono/default.nix:24]: See above

**Review by yoav-lavi — COMMENTED** [review]
Looks good, left one comment I wasn't sure about, will approve once we discuss it

File: pkgs/data/fonts/commit-mono/default.nix (hunk 1 above)
Comment (zzzsyyy) [reply pkgs/data/fonts/commit-mono/default.nix:23]: this seems to be the structure of upstream's zip file

<details>
<summary>Details</summary>

```
> ouch l CommitMono-1.138.zip --tree
Archive: ~/CommitMono-1.138.zip
└── CommitMono-1.138
   ├── CommitMono-400-Italic.otf
   ├── custom-settings.json
   ├── CommitMono-700-Regular.otf
   ├── CommitMono-400-Regular.otf
   ├── ttfautohint
   │  ├── CommitMono-400-Regular.ttf
   │  ├── CommitMono-700-Regular.ttf
   │  ├── CommitMono-400-Italic.ttf
   │  └── CommitMono-700-Italic.ttf
   ├── installation.txt
   ├── license.txt
   └── CommitMono-700-Italic.otf
```

</details>

**Review by yoav-lavi — APPROVED** [review]

**Pull request merged**